pkg/services/ngalert/api/api_trans_dev.go

```diff
@@ -131,3 +136,140 @@ func (api *API) conditionOldEndpointByID(c *models.ReqContext) response.Response
 
 	return response.JSON(200, evalCond)
 }
+
+// conditionEvalEndpoint handles POST /api/alert-definitions/ruleGroupByOldID.
+func (api *API) ruleGroupByOldID(c *models.ReqContext) response.Response {
+	id := c.ParamsInt64("id")
+	if id == 0 {
+		return response.Error(400, "missing id", nil)
+	}
+
+	save := c.Query("save") == "true"
+
+	getAlert := &models.GetAlertByIdQuery{
+		Id: id,
+	}
+
+	if err := bus.Dispatch(getAlert); err != nil {
+		return response.Error(400, fmt.Sprintf("could find alert with id %v", id), err)
+	}
+
+	if getAlert.Result.OrgId != c.SignedInUser.OrgId {
+		return response.Error(403, "alert does not match organization of user", nil)
+	}
+
+	oldAlert := getAlert.Result
+
+	sb, err := oldAlert.Settings.ToDB()
+	if err != nil {
+		return response.Error(400, "failed to marshal alert settings", err)
+	}
+
+	evalCond, err := translate.DashboardAlertConditions(sb, c.OrgId)
+	if err != nil {
+		return response.Error(400, "Failed to translate alert conditions", err)
+	}
+
+	getDash := &models.GetDashboardQuery{
+		Id:    getAlert.Result.DashboardId,
+		OrgId: getAlert.Result.OrgId,
+	}
+	if err := bus.Dispatch(getDash); err != nil {
+		return response.Error(400, fmt.Sprintf("could find dashboard %v for alert with id %v", getDash.Id, id), err)
+	}
+
+	isGeneralFolder := getDash.Result.FolderId == 0 && !getDash.Result.IsFolder
+
+	var namespaceUID string
+
+	if isGeneralFolder {
+		namespaceUID = "General"
```

Comment:
A dashboard can have permissions [directly assigned to it](https://github.com/grafana/grafana/blob/87c3a2b7904d8a3ece112e1189063f2c8db03223/pkg/services/sqlstore/dashboard_acl.go#L79) (not only inherits the permissions of its parent folder if exists). So I think that in this case a new folder should be created with these permissions.
`UpdateRuleGroup()` will be modified and it will create the namespace if it does not exist but we have to pass those permissions. So I think we need to modify `UpdateRuleGroupCmd` and optionally pass them there WDYT?

Reply:
Sounds like a step in the right direction :+1: 

I wonder what we would expect it to do if it gets permissions, and is not creating a folder.